client/src/components/DisplayTable/DisplayTableUtils.js

```diff
@@ -62,3 +63,204 @@ export const DisplayTableDownloadCsv = ({ csv_string, table_name }) => (
+          1,
+          _.first(raw_page_options) - 1,
+          ...raw_page_options,
+          _.last(raw_page_options) + 1,
+          page_count,
+        ];
+      } else {
+        if (!_.includes(raw_page_options, 1)) {
+          return [1, _.first(raw_page_options) - 1, ...raw_page_options];
+        } else if (!_.includes(raw_page_options, page_count)) {
+          return [
+            ...raw_page_options,
+            _.last(raw_page_options) + 1,
+            page_count,
+          ];
+        }
+      }
+    } else {
+      return raw_page_options;
+    }
+  };
+
+  const page_options = get_page_options(num_pages, current_page + 1);
+  return (
+    <tr className="page-selector-controls">
+      <td colSpan={num_col}>
+        <div style={{ padding: "10px" }}>
+          <div className="frow">
+            <div
+              className="fcol-xs-12 d-flex justify-content-center page-selector"
+              style={{ flexWrap: "wrap" }}
+            >
+              {_.map(page_options, (page_num, index) => (
+                <button
+                  key={page_num}
+                  className={`btn-ib-light${
+                    (page_num - 1 === current_page && "--reversed") || ""
+                  }`}
+                  onClick={() => change_page(page_num - 1)}
+                  aria-label={text_maker("page", { page_num: page_num })}
+                  role="tab"
+                  aria-selected={`${page_num - 1 === current_page}`}
+                  dangerouslySetInnerHTML={{
+                    __html:
+                      index !== 0 &&
+                      index !== page_options.length - 1 &&
+                      (page_options[index + 1] !== page_num + 1 ||
+                        page_options[index - 1] !== page_num - 1)
+                        ? "&hellip;"
+                        : page_num,
+                  }}
+                ></button>
+              ))}
+            </div>
+          </div>
+        </div>
+      </td>
+    </tr>
+  );
+};
+
+export const SelectPageSize = ({
+  selected,
+  on_select,
+  page_size_increment,
+  num_items,
+  num_options_max,
+}) => {
+  const num_options =
+    _.ceil(num_items / page_size_increment) > num_options_max
+      ? num_options_max
+      : _.ceil(num_items / page_size_increment);
+
+  const options = _.chain(num_options + 1)
+    .range()
+    .map((_, ix) => {
+      if (ix === num_options) {
+        return { value: num_items, label: text_maker("show_all") };
+      } else {
+        return {
+          value: (ix + 1) * page_size_increment,
+          label: (ix + 1) * page_size_increment,
+        };
+      }
+    })
+    .value();
+
+  const dropdown_content = (
+    <form className="paginate_by_dropdown">
+      <div style={{ marginBottom: 10 }}>
+        {_.map(options, ({ label, value }) => (
+          <div key={value} onClick={() => on_select(value)}>
+            <input
+              type={"radio"}
+              name={"paginate_by_dropdown"}
```

Comment:
So radio buttons are grouped with `name` I believe, so if there were two paginated tables on the same page, I believe this radio button won't work. I think just giving each instances a unique name should fix it

Reply:
So I just gave it a quick test, it seems like everything continues to work fine

![image](https://user-images.githubusercontent.com/25855114/112487054-d082cd80-8d52-11eb-8d0c-015571ecffb1.png)


Should I add a name tag anyways?

Reply:
Probably currently safe because the pagination options menu closes on click-out. Can't open two different table's menus at the same time. If that menu closing behaviour changed though, this would likely become a bug. Worth being safe/not leaving land mine bugs around, I'd say make them unique (see lodash's `_.uniqueId` util).